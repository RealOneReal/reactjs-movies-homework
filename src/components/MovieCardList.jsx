import { Grid } from "@mui/material";
import MovieCard from "./MovieCard";

const MovieCardList = (props) => {
  const { movieCards } = props;
  return (
    <Grid container>
      {movieCards.map((card) => (
        <Grid item xs={9} md={2} key={card.id}>
          <MovieCard props={card} />
        </Grid>
      ))}
    </Grid>
  );
};
export default MovieCardList;
