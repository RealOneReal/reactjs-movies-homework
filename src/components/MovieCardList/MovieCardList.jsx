import { Grid } from "@mui/material";
import MovieCard from "../MovieCard/MovieCard";

const MovieCardList = (props) => {
  const { results } = props;
  return (
    <Grid container spacing={1} justifyContent="center">
      {results.map((card) => (
        <Grid item key={card.id}>
          <MovieCard {...card} />
        </Grid>
      ))}
    </Grid>
  );
};
export default MovieCardList;
