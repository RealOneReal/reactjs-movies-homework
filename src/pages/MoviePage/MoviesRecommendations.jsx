import MovieCardList from "../../components/MovieCardList/MovieCardList";
import DataFromTMDB from "../../mock-data/mockMovies";

const MoviesRecommendations = () => {
  // todo: get recommendations from Redux ,it`s temporary implementation

  const results = DataFromTMDB?.results.slice(0, 5);
  return <MovieCardList {...{ results }} />;
};

export default MoviesRecommendations;
