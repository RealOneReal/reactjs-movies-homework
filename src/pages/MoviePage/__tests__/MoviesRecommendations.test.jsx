import { render } from "@testing-library/react";
import MoviesRecommendations from "../MoviesRecommendations";
import MovieCardList from "../../../components/MovieCardList/MovieCardList";

jest.mock(
  "../../../components/MovieCardList/MovieCardList",
  () => () => "MovieCardList"
);

describe("MoviesRecommendations", () => {
  it("should be rendered", () => {
    const { container } = render(<MoviesRecommendations />);
    expect(container).toMatchSnapshot();
  });
});
