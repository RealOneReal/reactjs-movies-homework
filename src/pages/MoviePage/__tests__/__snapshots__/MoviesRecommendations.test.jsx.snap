// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesRecommendations should be rendered 1`] = `
<div>
  MovieCardList
</div>
`;
